tests/test_differentiation.py

```diff
@@ -65,6 +65,52 @@ def test_adjoint_diff(n_qubits: int, n_layers: int) -> None:
     # )
 
 
+@pytest.mark.parametrize("n_qubits", [3, 5])
+@pytest.mark.parametrize("batch_size", [1, 2])
+@pytest.mark.parametrize("ops_op", [pyq.Z, pyq.Y])
+@pytest.mark.parametrize("dtype", [torch.complex128])
+def test_sampled_diff(
+    n_qubits: int,
+    batch_size: int,
+    ops_op: Primitive,
+    dtype: torch.dtype,
+) -> None:
+    rx = pyq.RX(0, param_name="theta_0")
+    cry = pyq.CPHASE(0, 1, param_name="theta_1")
+    rz = pyq.RZ(2, param_name="theta_2")
+    cnot = pyq.CNOT(1, 2)
+    ops = [rx, cry, rz, cnot]
+    circ = pyq.QuantumCircuit(n_qubits, ops).to(dtype)
+    obs = pyq.Observable(pyq.Add([ops_op(i) for i in range(n_qubits)])).to(dtype)
```

Comment:
You can also try to use the `random_pauli_hamiltonian` in `helpers.py`.

Reply:
Using `random_pauli_hamiltonian`, the tests fail with hamiltonians composed of both 1q and 2q terms. Investigating this

Reply:
This is due to the error scaling with the observable definition as well as the number of shots. Changing the test now for testing simpler cases of using coeffs 1.0 with Scale in gpsr test.